python/foglamp/services/core/api/service.py

```diff
@@ -95,42 +96,51 @@ def get_service_records():
         is_enabled = True if ((type(enabled) is str and enabled.lower() in ['t', 'true']) or (
             (type(enabled) is bool and enabled is True))) else False
 
-        storage = connect.get_storage_async()
+        # ---------------------------------------- Sanity checks
+        # Check if a valid plugin has been provided
+        try:
+            # "plugin_module_path" is fixed by design. It is MANDATORY to keep the plugin in the exactly similar named
+            # folder, within the plugin_module_path.
+            plugin_module_path = "foglamp.plugins.south" if service_type == 'south' else \
+                                 "foglamp.plugins.north" if service_type == 'north' else None
+            import_file_name = "{path}.{dir}.{file}".format(path=plugin_module_path, dir=plugin, file=plugin)
+            _plugin = __import__(import_file_name, fromlist=[''])
+        except ImportError as ex:
+            raise web.HTTPInternalServerError(reason='Plugin "{}" import problem from path "{}". {}'.format(plugin, plugin_module_path, str(ex)))
+        except Exception as ex:
+            raise web.HTTPInternalServerError(reason='Failed to create plugin configuration. {}'.format(str(ex)))
+
+        storage = connect.get_storage()
 
         # Check that the process name is not already registered
         payload = PayloadBuilder().SELECT("name").WHERE(['name', '=', name]).payload()
-        result = await storage.query_tbl_with_payload('scheduled_processes', payload)
+        result = storage.query_tbl_with_payload('scheduled_processes', payload)
         count = result['count']
         if count != 0:
             raise web.HTTPBadRequest(reason='A service with that name already exists')
 
         # Check that the schedule name is not already registered
         payload = PayloadBuilder().SELECT("schedule_name").WHERE(['schedule_name', '=', name]).payload()
-        result = await storage.query_tbl_with_payload('schedules', payload)
+        result = storage.query_tbl_with_payload('schedules', payload)
         count = result['count']
         if count != 0:
             raise web.HTTPBadRequest(reason='A schedule with that name already exists')
+        # ----------------------------------------
 
-        # First create the scheduled process entry for our new service
-        if service_type == 'south':
-            script = '["services/south"]'
-            plugin_module_path = "foglamp.plugins.south"
-        if service_type == 'north':
-            script = '["services/north"]'
-            plugin_module_path = "foglamp.plugins.north"
+        # Now first create the scheduled process entry for the new service
+        script = '["services/south"]' if service_type == 'south' else \
+                 '["services/north"]' if service_type == 'north' else None
         payload = PayloadBuilder().INSERT(name=name, script=script).payload()
```

Comment:
can script be None? better to check `if service_type.lower() is not in  [south, north]` then raise typeerror; else script = '["services/{}"]'.format(service_type.lower())